Library/Homebrew/utils/popen.rb

```diff
@@ -12,8 +12,25 @@ def self.safe_popen_read(*args, **options, &block)
     raise ErrorDuringExecution.new(args, status: $CHILD_STATUS, output: [[:stdout, output]])
   end
 
-  def self.popen_write(*args, **options, &block)
-    popen(args, "wb", options, &block)
+  def self.popen_write(*args, **options)
+    popen(args, "w+b", options) do |pipe|
+      output = ""
+
+      # Before we yield to the block, capture as much output as we can
+      loop do
+        output += pipe.read_nonblock(4096)
```

Comment:
Sorry, last thing: why 4096? Can/should this be a variable or comment or the default?

Reply:
That’s the standard buffer size used throughout the [API documentation of Ruby’s IO class](https://docs.ruby-lang.org/en/2.6.0/IO.html).

It doesn’t really matter which (reasonable) buffer size we use here. We just need to specify one because Ruby doesn’t offer a default for `#read_nonblock`.